torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -313,14 +313,14 @@ def call_function(
         # ops - see torch/dispatch/_dispatcher.py
         if len(args) != 4:
             raise UserError(
-                UserErrorType.DYNAMIC_CONTROL_FLOW,
+                UserErrorType.GRAPH_BREAK_IN_CONTROL_FLOW,
```

Comment:
I expected us to have a giant try-catch around the call_function that could raise UncapturableHigherOrderOpError, or some sort of decorator to make it nicer. The error message should say something like "cond doesn't work unless it is captured with torch.compile" and we should rethrow the original error